lakefs config.yaml align name and search order (#2372)
treeverse/lakeFS: docs/reference/configuration.md

{% include toc.html %}

Configuring lakeFS is done using a yaml configuration file and/or environment variable.

> The configuration file location can be set with the '--config' flag. It will lookup, load, and use the first file found in the following order:
> 1. ./config.yaml
> 1. `<home directory>`/lakefs/config.yaml
> 1. /etc/lakefs/config.yaml
> 1. `<home directory>`/.lakefs.yaml

Reviewer: This confused me because `--config` override the existing path.  Consider
```suggestion
The configuration file location can be set with the '--config' flag. If not specified, the the first file found in the following order will be used:
1. ./config.yaml
1. `$HOME`/lakefs/config.yaml
1. /etc/lakefs/config.yaml
1. `$HOME`/.lakefs.yaml
```
